core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -134,6 +138,14 @@ public final String getPath() {
     return path;
   }
 
+
+  /**
+   * Set MetricsContext labels.
+   */
+  protected void setupMetricsContext(T appConfig) {
```

Comment:
Thinking out loud here - since this is protected, a developer can assume that calling this after initialization would update the MetricsContext in Metrics object but that is not the case. Maybe we can remove this altogether or make it private

Reply:
Ah, I am wrong

Reply:
We could defer Metrics initialization to `configureHandler` so it's more similar to the pattern used by `setupResources`, `configureSecurityHandler` etc. I'm not sure there is any good reason you start the metric reporters before you can handle any traffic. That assumption may be incorrect fo something like C3 though.

Reply:
That would be ideal but we do need Metrics object before we start the Application in C3. Dependency is bit convoluted in C3. For now, let keep this the way it is until we can fully understand the impact on downstream projects.

Reply:
I'll add a docstring for now

Reply:
I just went ahead and moved everything into the config. It feels a bit weird as MetricContext labels don't really feel configurable like MetricLabels but they are propagated as configs to downstream components so it is what it is.